include/gsl.h

```diff
@@ -122,15 +122,15 @@ class not_null
     not_null<T>& operator=(std::nullptr_t) = delete;
 	not_null<T>& operator=(int) = delete;
     
-    T get() const {
-#ifdef _MSC_VER
-        __assume(ptr_ != nullptr);
-#endif
-        return ptr_;
-    } // the assume() should help the optimizer
-
-    operator T() const {  return get(); }
-    T operator->() const { return get(); }
+    // if T has .get() forward its value wrapped in not_null
+    template<typename U = T, typename = decltype(std::declval<U>().get())>
```

Comment:
In some cases, not naming a template parameter causes a problem for MSVC 2013, so the usage probably spread further than it needed to. If you are making changes and don't want the name and it doesn't cause a target compiler a problem....then feel free to do without it.